feat: when html theme changes, mermaid theme updates accordingly

fresto32 opened the pull request:
Closes #4 

The proposed pull request offers the following changes:
1. Use a `MutationObserver` to watch for changes to `html.data-theme` to update mermaid config accordingly
2. Uses the mermaid api's `render` function, since it is the recommended way for integrating `mermaid` in a webpage (see: [docs](https://github.com/mermaidjs/mermaidjs.github.io/blob/master/docs/mermaidAPI.md#mermaidapi))

Thanks for your work in developing this. I really like mermaid and can't imagine not being able to use it in Docusaurus.

Example:

![mermaid-pr](https://user-images.githubusercontent.com/13295571/147711258-c59640c8-6bbd-4ec2-93b1-3ed58930a390.gif)

I refresh the page a number of times in the gif - seen when the page turns white.

sjwall commented on src/Mermaid.tsx:
```diff
@@ -33,12 +40,40 @@ export type MermaidProps = {
+  const html: HTMLHtmlElement = document.querySelector('html')!
+
+  // Watch for changes in theme in the HTML attribute `data-theme`.
+  const [theme, setTheme] = useState<mermaidAPI.Theme>('default')
```
The initial theme needs to be based on the data-theme attribute otherwise when the page is loaded in dark theme it will be in the light theme

fresto32 replied:
Whoops! Nice catch. I've moved all the theme logic to a new file called `theme.helper.spec.ts`

sjwall commented on src/Mermaid.tsx:
```diff
@@ -33,12 +40,40 @@ export type MermaidProps = {
+        continue
+      }
+
+      setTheme(html.getAttribute('data-theme') === 'dark' ? 'dark' : 'default')
```
Can the theme used be added as config options or a callback defined in config.model.ts?

fresto32 replied:
Yup - great idea! I've added it to the config model

fresto32 commented:
Thanks for the review Sam! I've addressed some of the concerns :) . I'd like to improve the test coverage and double check everything. I'll do that tomorrow

fresto32 commented:
> Thanks for the review Sam! I've addressed some of the concerns :) . I'd like to improve the test coverage and double check everything. I'll do that tomorrow

All done. Added some more tests in https://github.com/sjwall/mdx-mermaid/pull/21/commits/d527a1684e45e922d4955a37397f7fa81bb020bc

sjwall commented:
Thank you for the PR, it has been released in `v1.2.0`
